From 0ceaf99bf18517bf76eb0cd4400a569daa0224ec Mon Sep 17 00:00:00 2001
From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Mon, 5 Jul 2010 17:04:38 +0200
Subject: [PATCH 3/3] make dbus-1.pc relocatable

Don't use expanded paths in dbus-1.pc.in

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
# 20100825 wsa: according to mol not necessary for mainline.

 configure.ac |    3 +++
 dbus-1.pc.in |    6 +++---
 2 files changed, 6 insertions(+), 3 deletions(-)

Index: dbus-1.4.1/configure.ac
===================================================================
--- dbus-1.4.1.orig/configure.ac
+++ dbus-1.4.1/configure.ac
@@ -1510,10 +1510,13 @@ AC_DEFINE_UNQUOTED(DBUS_DATADIR,"$DBUS_D
 #### Directory to install dbus-daemon
 if test -z "$with_dbus_daemondir" ; then
     DBUS_DAEMONDIR=$EXPANDED_BINDIR
+    RELATIVE_DBUS_DAEMONDIR=$bindir
 else
     DBUS_DAEMONDIR=$with_dbus_daemondir
+    RELATIVE_DBUS_DAEMONDIR=$with_dbus_daemondir
 fi
 AC_SUBST(DBUS_DAEMONDIR)
+AC_SUBST(RELATIVE_DBUS_DAEMONDIR)
 AC_DEFINE_UNQUOTED(DBUS_DAEMONDIR,"$DBUS_DAEMONDIR", [Directory for installing the DBUS daemon])
 
 #### Directory to install the other binaries
Index: dbus-1.4.1/dbus-1.pc.in
===================================================================
--- dbus-1.4.1.orig/dbus-1.pc.in
+++ dbus-1.4.1/dbus-1.pc.in
@@ -3,11 +3,11 @@ exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
 system_bus_default_address=@DBUS_SYSTEM_BUS_DEFAULT_ADDRESS@
-sysconfdir=@EXPANDED_SYSCONFDIR@
-session_bus_services_dir=@EXPANDED_DATADIR@/dbus-1/services
+sysconfdir=@sysconfdir@
+session_bus_services_dir=@datadir@/dbus-1/services
 system_bus_services_dir=@EXPANDED_DATADIR@/dbus-1/system-services
 interfaces_dir=@EXPANDED_DATADIR@/dbus-1/interfaces
-daemondir=@DBUS_DAEMONDIR@
+daemondir=@RELATIVE_DBUS_DAEMONDIR@
 
 Name: dbus
 Description: Free desktop message bus
